web/src/pages/billing/SeqrProportionalMapGraph.tsx

```diff
@@ -0,0 +1,208 @@
+                                        [projectObj.project]: projectObj.percentage,
+                                    }),
+                                    {}
+                                ),
+                            },
+                        })),
+                    }),
+                    {} as { [tMethod in ProportionalDateTemporalMethod]: IStackedAreaChartData[] }
+                )
+
+                // convert end to date if exists, or use current date
+                const graphEnd = !!end ? new Date(end) : new Date()
+
+                // If the graph isn't at endDate, add a second entry at the endDate
+                // so it finishes at endDate for a better visual graph
+                for (const temporalMethod of Object.keys(allGraphData)) {
+                    const temporalMethodKey = temporalMethod as ProportionalDateTemporalMethod
+                    const values = allGraphData[temporalMethodKey]
+                    const lastValue = values[values.length - 1]
+                    if (lastValue.date <= graphEnd) {
+                        allGraphData[temporalMethodKey].push({
+                            date: graphEnd,
+                            values: lastValue.values,
+                        } as IStackedAreaChartData)
+                    }
+                }
+
+                setAllPropMapData(allGraphData)
+            })
+            .catch((er: Error) => {
+                // @ts-ignore
+                let message = er.response?.data?.description || er?.message
+                setError(message)
+                setIsLoading(false)
+            })
+    }
+
+    function getSeqrProjects() {
+        const api = new ProjectApi()
+        api.getSeqrProjects()
+            .then((projects) => {
+                const newProjects = projects.data.reduce(
+                    (prev: { [project: string]: boolean }, project: Project) => ({
+                        ...prev,
+                        [project.name!]: true,
+                    }),
+                    {}
+                )
+                updateProjects(newProjects)
+            })
+            .catch((er) => {
+                setError(er.message)
+                setIsLoading(false)
+            })
+    }
+
+    // on first load
+    React.useEffect(() => {
+        if (!projectSelections) {
+            getSeqrProjects()
+        }
+    }, [])
+
+    React.useEffect(() => {
+        if (projectSelections) {
+            loadPropMap(projectSelections)
+        }
+    }, [start, end])
+
+    if (!allPropMapData && !isLoading && !error) {
+        return <Message negative>No data found</Message>
+    }
+
+    const selectedProjects: string[] = projectSelections
+        ? _.sortBy(Object.keys(projectSelections).filter((project) => projectSelections[project]))
+        : []
+
+    let graphComponent: React.ReactElement | undefined = undefined
+
+    if (allPropMapData) {
```

Comment:
It's probably neater to put a catch in the StackedAreaChart to return an empty fragment if there is no data

Reply:
Yeah good call, actioned :)

Reply:
Good shout, have actioned this.